meson.build

```diff
@@ -16,6 +16,14 @@ mandir = get_option('mandir')
 lutrisdir = join_paths(pylibdir, 'lutris')
 podir = join_paths(meson.source_root(), 'po')
 
+dependency('gio-2.0')
+dependency('glib-2.0')
+dependency('gobject-introspection-1.0')
+dependency('gtk+-3.0')
+dependency('webkit2gtk-4.0')
+dependency('gnome-desktop-3.0', required: false)
+dependency('appindicator-0.1', required: false)
```

Comment:
just having dependencies doesn't do anything.
What consumes these?

Reply:
Meson obviously, it is meant as a hint that you are missing a dependency. This is specially important with python apps, when you are not going to crash until runtime.

I maintain the flatpak version of the app and I just realized it used libnotify while randomly browsing the code, I had to patch the flatpak and send a patch here too, I would have send those patches earlier if I knew it was used.

Reply:
yeah but those are the low level dependencies, not the python dependencies.
gtk3 being available does not mean its available in python.

Also since no one consumes these and are only checked against, aren't the dependencies that aren't required obsolete?

Perhaps a better solution would be to generate python code on the fly and check if it can import the required modules

Reply:
> aren't the dependencies that aren't required obsolete?

No, libappindicator is used optionally for example.

>  gtk3 being available does not mean its available in python.

if the gobject instrospection dep is satisfied and then having gtk3 means having gtk3 in python.

> Perhaps a better solution would be to generate python code on the fly and check if it can import the required modules

this is already the case, but it is simpler if packagers had a way to determine which dependencies they need, which is what this helps with.

Reply:
> No, libappindicator is used optionally for example.

Yeah but the actual dependency is not consumed by anything, thus this is used for nothing more than a fancy message and anything that is not required is essentially pointless.

> if the gobject instrospection dep is satisfied then having gtk3 means having gtk3 in python.

right, bad example since the python gobject bindings enable any gobject based library to be used from within python.

Lets take two different examples which are not covered by this:

- libmagic (dynamically loaded [here](https://github.com/lutris/lutris/blob/master/lutris/util/magic.py#L209))
    *this* would fit into this dependency check, but since there are many different names that are checked against (and checking against multiple names would require bumping the required meson version even more) its not feasible to do this
- PyYAML
    technically speaking this depends on libyaml, but we cannot ensure that if libyaml is present pyyaml is also

> this is already the case, but it is simpler if packagers had a way to determine which dependencies they need, which is what this helps with.

How dependencies are structured out, packaged and named differs greatly between distros (and operating systems) so packagers would still need to do some amount of digging around.

But ignoring that, these dependencies generally look for files used for development with those libraries using pkg-config and cmake, but none of which are required to run the python modules.

Taking the Fedora package as an example, this would require adding more dependencies (especially -devel dependencies) that aren't needed to build lutris for effectively no gain

Here is a practical example for gobject

Before this PR
```spec
BuildRequires:  python3-gobject # ropes in the gobject-introspection
```

After this PR
```spec
BuildRequires:  python3-gobject, gobject-introspection-devel
```

Reply:
You are right, forgot about that. Removed. Its such a shame thats not possible to declare dependencies in python in any sane way in meson.

Reply:
Find_installation() can check for `modules: []` that must be able to be imported, so there is that. This could help for pyyaml although gobject is a bit more complicated...